artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -664,7 +664,7 @@ public void addTail(final MessageReference ref, final boolean direct) {
                if (intermediateMessageReferences.isEmpty() && messageReferences.isEmpty() && !pageIterator.hasNext() && !pageSubscription.isPaging()) {
                   // We must block on the executor to ensure any async deliveries have completed or we might get out of order
                   // deliveries
-                  if (flushExecutor() && flushDeliveriesInTransit()) {
+                  if (internalFlushExecutor(500, false) && flushDeliveriesInTransit()) {
```

Comment:
Why 500 why not 100, if the issue was previously it was taking too long surely we want this as small as safely possible?